### PR TITLE
Feat/generalized pow

### PR DESCRIPTION
**Pull request by paulhandy**

### CODE REVIEW

**Comment by paulhandy** [comment]
Not yet ready for merge.

File: checksum/src/checksum.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+use hamming::weight;
+use num::BigUint;
+
+const identity: [usize; mem::size_of::<usize>()] = (0..mem::size_of::<usize>())
```
Comment (th0br0) [inline checksum/src/checksum.rs:4]: size_of returns size in bytes, you want bits though.

File: checksum/src/checksum.rs [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    .iter()
+    .map(|i| 1 << i)
+    .collect();
+
+pub fn transpose(state: &[BCTrit]) {
+    let mut output: [(BigUint, BigUint); mem::size_of::<usize>()];
+    let bytes: [u8; state.len()] = [0u8; state.len()];
+    for i in 0..mem::size_of::<usize>() {
```
Comment (th0br0) [inline checksum/src/checksum.rs:12]: As above, size_of is bytes not bits

File: checksum/src/checksum.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
+use hamming::weight;
+use num::BigUint;
+
+const identity: [usize; mem::size_of::<usize>()] = (0..mem::size_of::<usize>())
+    .iter()
+    .map(|i| 1 << i)
+    .collect();
+
+pub fn transpose(state: &[BCTrit]) {
+    let mut output: [(BigUint, BigUint); mem::size_of::<usize>()];
```
Comment (th0br0) [inline checksum/src/checksum.rs:10]: Why biguint? the `from_bytes_le` is also available for usize

Comment (paulhandy) [reply checksum/src/checksum.rs:10]: because the word size is as long as 243-bit

**Pull request closed without merging**